DigitalExcellence/dex-backend
# Bug/user bugs fix

## Description

This pull request entails all the found bugs:
- GetHighlight had the parameter projectid instead of highlight id.
- The user was allowed to delete the admin and registereduser role, this is now disallowed.
- Create user did not have a role, this now has the default registereduser role.
- Removed the unneeded linkedserviceresource inside the userResource.
- Get highlight by id did not return a project.
- Highlight repository did not redact user information.
- Added back the develop client id / secret.
- Changed the roles endpoint to role.
- User can now call delete on its own user id.
- Added user self delete endpoint.
- Removed IIS from launchsettings.
- UserExtension checks for client_role backendapplication.
- Added backendrole to the identity server api client.
- Fixed typos in error message.
- Disabled the configure await and redundant whiteline(resharper still gives the warnings).
- Added logging on exceptions.
- Modified the project user to only have limited information(id, name, email).
- Changed exception to more descriptive exception.
- Disabled warnings for the identity callback function because this is a todo for later.
- Added role id to rolescope for ef core.
- Removed some possible null ref exception flows.
- Added a project to the highlight generator.
- Made Program class static.
- Fixed broken role update.
- Added guid validation in the get embedded project.
- Removed unused statistics repository.
- Checked for unauthorizedaccess exception to change the 500 error to 401 unauthorized.
- Added embed scopes to PR role.


## Type of change


## Checklist


## Steps to Test or Reproduce

to test if the bugs are fixed run the postman tests.
see issue #178 for full descriptions

## Link to issue

Closes: #168 
Closes: #154 
Closes: #105 
Closes: #178
